instruqt/protecting-your-assets/06-validating-license-fields/assignment.md

```diff
@@ -0,0 +1,188 @@
+Application with
+Replicated](https://play.instruqt.com/replicated/tracks/distributing-with-replicated)
+lab.
+
+Interacting with the SDK
+========================
+
+Since we're going to shell commands to call the SDK, let's use the "Netshoot"
+image from Nicola Kabar. This image contains a lot of network troubleshooting
+tools and tools you need to work with them. That means it has `curl`, `jq`, and
+other commands that make it easier to work with APIs from the command line.
+
+```
+kubectl run -n slackernews sdk-client -it --image nicolaka/netshoot
+```
+
+Let's get the license fields for the Slackernews application.
+
+```
+curl http://replicated:3000/api/v1/license/fields | jq .
+```
+
+This result will show the expiration date and the field you added in the
+previous step of the lab.
+
+```
+{
+  "expires_at": {
+    "name": "expires_at",
+    "title": "Expiration",
+    "description": "License Expiration",
+    "value": "",
+    "valueType": "String",
+    "signature": {
+      "v1": "HlS3RIiNVGLqijbt2fvbTJyvXKNRxiMqlhS6FkJDL80NprYIc8pRQxfJRj15QtN85bJrybcqgJQfeG3mto29+EFzdic/mGhqiEp8NbMGQDMAM1UrSR7OY/9g38ljogUAUx0I9KOpdDVeXSJEAyw5GxHFrjtXYjb+5Zc4Ed4WxijmfQsUbBHQ4qWqV84x0BYAzoQ2VRiIjt7B5tWDJ2h3j8zExE98qt5RrxcaGe7Ori/6dFVUYVCUg2InCYN5oR5CiJMTgmFdgmAje368JJHVny/jxsqEd+od9rvhgF0r8ksqlR9FpzdYB4L2eA4jQd4qeJK+v7DoDNKmQCd+hJbKDA=="
+    }
+  },
+  "member_count_max": {
+    "name": "member_count_max",
+    "title": "Member Count",
+    "value": 100,
+    "valueType": "Integer",
+    "signature": {
+      "v1": "d4oAFJ8trT1SVwnDa5xkURT25TD3gQwaS/urGMaAC4vOx+nKItEgI8jLySyM4ST6X+7MkJ4r4Sb9qpL9+JzeSkZA+pNGqZonebg7MaLHme3sAmw7v4+TPn+MO53fdEnVPhsZj33m6EUQvhORhxOQlVqwjVjlTbHBWidmnUaQkUr7RXLPhVL6RQJxwde6yaObifxY0Ux2BL6nbxTIwMh/yBchx8GR+gHyi6MdEBbT80P3AxzOAVPflpK2y2eX84b8wST9zHel4dL+FTPftXdqVbJB/99TIFmeiKONCi7JMq+6rGiXqgOgPg0h5YVgiS1UlLML47IrBT74Rq+EGnAE6g=="
+    }
+  }
+}
+```
+
+To get a single field, add the field name to the URI.
+
+```
+curl http://replicated:3000/api/v1/license/fields/member_count_max | jq .
+```
+
+This will give you the value for the "Member Count" entitlement you added to
+the license earlier.
+
+```
+{
+  "name": "member_count_max",
+  "title": "Max Member Count",
+  "value": 100,
+  "valueType": "Integer",
+  "signature": {
+    "v1": "ttq5sqIQbY3U+LgbqEKMQuKn5UO5r3tvHnc18Cp1JPEUGX7f6X7ZjkKO4UkTHqlyC3ZFxVc0e6FVDH02+Qw4b9biPShFIPz0wzoWdGzzuNGWVfGUIQKXYR3WXLapKaaOeeJLM6c/Z90MzW9cwxBrUVbyAzaBagBPoegL0/PpKhM2NUKpDYeDHPRHnFgANXDE+5IoB3EG4lSBivkWnioAmWZIbVcHHZTZbE9Bwdh6Tmo/J3NUFcSBUjaS23vk5y8pVQ9otefxt5DVUvZb3r6RST/xzAOLKvk0tZExRb2X/CzdmiFJct3g9sJXPMzQe/RBPUUQFtbJa+K1aHKgyK4AmA=="
+  }
+}
+```
+
+When you call the SDK from your code you can access the value in the `value`
+field. Here in the shell we just change our `jq` expression a little bit.
+
+```
+curl http://replicated:3000/api/v1/license/fields/member_count_max | jq .value
+```
+
+Disconnect from the client pod for now. We'll use it again when we validate the
+signed value.
```

Comment:
Would be nice to show the `exit` command here. I for one hit ctrl-d immediately bookmarking the page, and then had to go 'oh right exit'